nomad/job_endpoint.go

```diff
@@ -712,7 +709,18 @@ func (j *Job) BatchDeregister(args *structs.JobBatchDeregisterRequest, reply *st
 		return err
 	}
 
+	// Commit job deregistering request via Raft
+	_, index, err := j.srv.raftApply(structs.JobBatchDeregisterRequestType, args)
+	if err != nil {
+		j.logger.Error("batch deregister failed", "error", err)
+		return err
+	}
+
+	// Populate the reply with job information
+	reply.Index = index
```

Comment:
not sure if this is actually needed but followed the pattern in `Dispatch` - I guess it's somewhat relevant if we exit early if we hit an error below - but I'm unsure of the ramifications.  Do `Dispatch` or `BatchDispatch` expect to be atomic operations?